include/s3select.h

```diff
@@ -2013,13 +2054,19 @@ class base_s3object
   int getMatchRow( std::string& result)
   {
     multi_values projections_resuls;
+
+    if (m_is_limit_on && m_processed_rows == m_limit)
```

Comment:
the limit-operation condition is out-of-loop which process each row (for all types of data-sources)
its should be placed differently